.circleci/config.yml

```diff
@@ -0,0 +1,123 @@
+version: 2
+
+workflows:
+  version: 2
+  kubeapps:
+    jobs:
+      - test_go
+      - test_dashboard
+      - build_chartrepo:
+          requires:
+            - test_go
+      - build_chartsvc:
+          requires:
+            - test_go
+      - build_chart_apprepository:
+          requires:
+            - test_go
+      - build_dashboard:
+          requires:
+            - test_dashboard
+      - build_tiller_proxy:
+          requires:
+            - test_go
+      - release:
+          requires:
+            - build_chartrepo
+            - build_chartsvc
+            - build_chart_apprepository
+            - build_dashboard
+            - build_tiller_proxy
+
+## Definitions
+install_gcloud_sdk: &install_gcloud_sdk
+  run: |
+    echo "export PATH=$PATH:${HOME}/google-cloud-sdk/bin" >> $BASH_ENV
+    echo "export CLOUDSDK_CORE_DISABLE_PROMPTS=1" >> $BASH_ENV
+    if [ ! -d ${HOME}/google-cloud-sdk/bin ]; then
+      rm -rf $HOME/google-cloud-sdk;
+      curl https://sdk.cloud.google.com | bash;
+    fi
+install_kubecfg: &install_kubecfg
+  run: |
+    if ! which kubecfg; then
+      wget -O kubecfg https://github.com/ksonnet/kubecfg/releases/download/v0.7.2/kubecfg-$(go env GOOS)-$(go env GOARCH)
+      install -m 755 kubecfg $GOPATH/bin/kubecfg
+    fi
+exports: &exports
+  run: |
+    # It is not possible to resolve env vars in the environment section:
+    # https://circleci.com/docs/2.0/env-vars/#using-bash_env-to-set-environment-variables
+    echo "export VERSION=${CIRCLE_TAG:-build-$CIRCLE_WORKFLOW_ID}" >> $BASH_ENV
+build_image: &build_image
+  machine: true
+  working_directory: /home/circleci/.go_workspace/src/github.com/kubeapps/kubeapps
+  environment:
+    GOPATH: /home/circleci/.go_workspace
+  steps:
+    - checkout
+    - <<: *exports
+    - run: make VERSION="$VERSION" $IMAGE
+    - run: |
+        if [[ -z "$CIRCLE_PULL_REQUEST" ]]; then
+          docker login -u="${DOCKER_USERNAME}" -p="${DOCKER_PASSWORD}"
+          if [[ -n "$CIRCLE_TAG" ]]; then 
+            docker push $IMAGE:$VERSION
+          fi
+          if [[ "$CIRCLE_BRANCH" == "master"  ]]; then
+            docker tag $IMAGE:$VERSION $IMAGE:latest
+            docker push $IMAGE:latest
+          fi
+        fi
+###
+
+jobs:
+  test_go:
+    working_directory: /go/src/github.com/kubeapps/kubeapps
+    environment:
+      CGO_ENABLED: "0"
+    docker:
+      - image: circleci/golang:1.9 
+    steps:
+      - checkout
+      - <<: *exports
+      - <<: *install_kubecfg
+      - <<: *install_gcloud_sdk
```

Comment:
Actually, when running `make test` it still generates the kubeapps manifest:

```
KUBEAPPS_VERSION=${VERSION:-latest} ;\
	KUBECFG_JPATH=./manifests/lib:./manifests/vendor/kubecfg/lib:./manifests/vendor/ksonnet-lib \
	kubecfg show -V VERSION=$KUBEAPPS_VERSION manifests/kubeapps.jsonnet > static/kubeapps-objs.yaml
```

We need `kubecfg` until we remove that. Also the `gcloud_sdk` is needed to run the tests in `github.com/kubeapps/kubeapps/pkg/gke` which I believe we can remove as well once we get rid of the `kubeapps` cli.

Reply:
We probably should have removed that as part of this PR

Reply:
I wanted to avoid unrelated changes (strictly speaking) in this PR. Once we have a specific PR to remove everything related to the `kubeapps` CLI we can modify all the related Makefile tasks.